draft-ietf-quic-version-negotiation.md

```diff
@@ -194,92 +215,49 @@ server.
-notation from the "Notational Conventions" section of {{QUIC}}):
+`version_negotiation`. The contents of handshake version information are shown
+below (using the notation from the "Notational Conventions" section of
+{{QUIC}}):
 
 ~~~
-Client Handshake Version Information {
+Handshake Version Information {
   Currently Attempted Version (32),
-  Previously Attempted Version (32),
-  Received Negotiation Version Count (i),
-  Received Negotiation Version (32) ...,
-  Compatible Version Count (i),
-  Compatible Version (32) ...,
+  Other Versions (32) ...,
 }
 ~~~
-{: #fig-client-hvi title="Client Handshake Version Information"}
+{: #fig-hvi-format title="Handshake Version Information Format"}
 
 The content of each field is described below:
 
 Currently Attempted Version:
 
-: The version that the client is attempting to use. This field MUST be equal to
-the value of the Version field in the long header that carries this data.
+: This field SHALL be equal to the value of the `Version` field in the long
+header that carries this data.
 
-Previously Attempted Version:
+The contents of the `Other Versions` field depends on whether it is sent by the
+client or by the server.
 
-: If the client is sending this first flight in response to a Version
-Negotiation packet, this field contains the version that the client used in the
-previous first flight that triggered the version negotiation packet. If the
-client did not receive a Version Negotiation packet, this field SHALL be
-all-zeroes.
+Client-Sent Other Versions:
 
-Received Negotiation Version Count:
+: When sent by a client, the `Other Versions` field lists all the versions that
+this first flight is compatible with, ordered by descending preference. Note
+that the version in the `Currently Attempted Version` field MUST be included in
+this list to allow the client to communicate the currently attempted version's
+preference. Note that this preference is only advisory, servers MAY choose to
+use their own preference instead.
 
-: A variable-length integer specifying the number of Received Negotiation
-Version fields following it. If the client is sending this first flight in
-response to a Version Negotiation packet, the subsequent versions SHALL include
-all the versions from that Version Negotiation packet in order, even if they
-are not supported by the client (even if the versions are reserved). If the
-client has not received a Version Negotiation packet on this connection, this
-field SHALL be 0.
+Server-Sent Other Versions:
 
-Compatible Version Count:
+: When sent by a server, the `Other Versions` field lists all the
+Fully-Supported Versions of this server deployment, see {{server-terms}}. This
+field conveys Fully-Supported Versions instead of Supported Versions to allow
+scenarios where the client's first flight (before receiving VN) reached a
+different server than the client's subsequent flight (after receiving VN).
```

Comment:
You might need to expand on this, as it is a key concept.  

This might be worth an entire section that says something like:

The client might receive a Version Negotiation packet from a server instance that doesn't support some Partially-Supported Versions; if it does so, those Partially-Supported Versions won't be listed in the Version Negotiation packet.  As part of validating the server's Other Versions, the client verifies that it would not have chosen any of the versions the server listed.  If the client supports any Partially-Supported Version and would prefer that version, it will detect a downgrade attack.  To avoid this, the server does not advertise any version that is Partially-Supported.

(And then we need text about how to turn on new versions properly.  (1) Deploy code and accept connections; (2) turn on VN for that version; (3) turn on Other Versions for that version and enjoy downgrade protection.)

Reply:
Agreed, fleshed out the "Server Deployments of QUIC" section